pkg/codegen/schema.go

```diff
@@ -263,11 +264,19 @@ func GenerateGoSchema(sref *openapi3.SchemaRef, path []string) (Schema, error) {
 			return Schema{}, fmt.Errorf("error turning reference (%s) into a Go type: %s",
 				sref.Ref, err)
 		}
+		skipOptionalPointer := globalState.options.OutputOptions.PreferSkipOptionalPointer
+		if schema.Type == nil || slices.Contains(*schema.Type, openapi3.TypeObject) {
+			// For object references pointers are the only way to denote optional, in contrast
+			// to other types where the zero-value can be used for that.
```

Comment:
Where does this get triggered? I can't seem to work out where this may occur :thinking:

Reply:
Hmm, I actually don't remember anymore. If I spot anything later, I'll add it here (or in a discussion, or issue or whatever would be appropriate then :smile: )